### PR TITLE
parse-url: if input contains a point, suppose it's an url

### PR DESCRIPTION
**Pull request by vindarel**
so we go to the site.

the previous behavior was to search with the default search
engine. Typing "wikipedia.com" searched for wikipedia on
duckduckgo. Now we go to it. 

Yes ?

### CODE REVIEW

**Comment by alexanderbarbosa** [comment]
hoho, I was about to do just that! That's neat! :D

File: source/utility.lisp [hunk 1]
```diff
@@ -39,6 +46,8 @@ SLIME."
             ((probe-file input-url)
              (format nil "file://~a"
                      (uiop:ensure-absolute-pathname input-url *default-pathname-defaults*)))
+            ((str:contains? "." input-url)
+             (str:concat "http://" input-url))
```
Comment (jmercouris) [inline source/utility.lisp]: I think it should always concatenate https instead of http

Comment (Ambrevar) [reply source/utility.lisp]: Agreed.

**Comment by Ambrevar** [comment]
I'd choose a different strategy since it won't work with "foo.bar baz":
if input is not a URL, prepend with "https://" and test with puri if the result is a valid URL.  If so, load it.  If not, go on with the search.

**Comment by vindarel** [comment]
oh indeed. 

@alexanderbarbosa me first :)

**Comment by vindarel** [comment]
done and rebased the commits.

File: source/utility.lisp [hunk 2]
```diff
@@ -39,6 +46,14 @@ SLIME."
...
              (format nil "file://~a"
                      (uiop:ensure-absolute-pathname input-url *default-pathname-defaults*)))
+            ((str:contains? "." input-url)
+             (let* ((new-url (str:concat "https://" input-url))
+                    (is-uri (ignore-errors
+                              (puri:uri-p (puri:parse-uri new-url)))))
+               (if is-uri
+                   new-url
+                   (generate-search-query input-url (cdr default)))
+               ))
```
Comment (Ambrevar) [inline source/utility.lisp]: Hanging parens ;)

File: source/utility.lisp [hunk 3]
```diff
@@ -39,6 +46,14 @@ SLIME."
             ((probe-file input-url)
              (format nil "file://~a"
                      (uiop:ensure-absolute-pathname input-url *default-pathname-defaults*)))
+            ((str:contains? "." input-url)
```
Comment (Ambrevar) [inline source/utility.lisp]: We can remove this check, nah?  Testing if `is-uri` should be enough.

File: next.asd [hunk 4]
```diff
@@ -9,6 +9,7 @@
   :defsystem-depends-on ("trivial-features")
   :depends-on (:alexandria
                :cl-strings
+               :str
```
Comment (Ambrevar) [inline next.asd]: Can we get rid of `cl-strings` then?

Comment (jmercouris) [reply next.asd]: I believe cl-strings has some things that str does not have yet.

Comment (vindarel) [reply next.asd]: Indeed, I had a look, and the only blocking function is cl-strings:insert. cl-strings also has functions to change case, which we don't use. I'll improve `str` :) Also, we'll have to compose with Quicklisp's delay (annoying mostly for users who install from source).

File: source/utility.lisp (hunk 3 above)
Comment (vindarel) [reply source/utility.lisp]: \+1. I ended with a cleaner logic but a repetition:

```
            ((puri:uri-p (ignore-errors
                           (puri:parse-uri (str:concat "https://" input-url))))
             (str:concat "https://" input-url))
```

you'd prefer it at the first `let*` ?

Comment (Ambrevar) [reply source/utility.lisp]: I'm fine with both, it's a trivial repetition.  Feel free to merge :D

**Pull request merged**